internal/repository/postgresql/postgresqluser/user.go

```diff
@@ -0,0 +1,159 @@
+	"context"
+	"database/sql"
+	"errors"
+	phonenumber "github.com/basliqlabs/qwest-services/pkg/mobile"
+
+	"github.com/basliqlabs/qwest-services/internal/entity/userentity"
+	"github.com/basliqlabs/qwest-services/pkg/errmsg"
+	"github.com/basliqlabs/qwest-services/pkg/richerror"
+)
+
+func (d *DB) DoesUserNameWithPasswordExist(ctx context.Context, username string, password string) (bool, error) {
+	const op = "postgresqluser.DoesUserNameWithPasswordExist"
+	stmt, err := d.db.Conn().PrepareContext(ctx, `SELECT user_id FROM users WHERE username=$1 AND password_hash=$2`)
+	if err != nil {
+		return false, richerror.New(op).
+			WithKind(richerror.KindUnexpected).
+			WithMessage(errmsg.CantScanQueryResult).
+			WithError(err)
+	}
+
+	row := stmt.QueryRowContext(ctx, username, password)
+
+	userId := new(int)
+	err = row.Scan(userId)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return false, richerror.New(op).
+				WithKind(richerror.KindNotFound).
+				WithMessage(errmsg.NotFound).
+				WithError(err)
+		}
+		return false, richerror.New(op).
+			WithKind(richerror.KindUnexpected).
+			WithMessage(errmsg.CantScanQueryResult).
+			WithError(err)
+	}
+	return true, nil
+}
+
+func (d *DB) FindUserByMobile(ctx context.Context, mobile string) (userentity.UserWithPasswordHash, bool, error) {
+	const op = "postgresqluser.FindUserByMobile"
+
+	normalizedMobile, err := phonenumber.NormalizePhoneNumber(mobile)
+
+	if err != nil {
+		return userentity.UserWithPasswordHash{}, false, err
+	}
+
+	stmt, err := d.db.Conn().PrepareContext(ctx,
+		`SELECT username, email, password_hash FROM users WHERE mobile=$1`)
+
+	if err != nil {
+		return userentity.UserWithPasswordHash{}, false, richerror.New(op).
+			WithKind(richerror.KindUnexpected).
+			WithMessage(errmsg.CantScanQueryResult).
+			WithError(err)
+	}
+
+	row := stmt.QueryRowContext(ctx, normalizedMobile)
+	var user userentity.UserWithPasswordHash
+	err = row.Scan(&user.UserName, &user.Email, &user.PasswordHash)
+	user.Mobile = normalizedMobile
+
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return userentity.UserWithPasswordHash{}, false, nil
+		}
+		return userentity.UserWithPasswordHash{}, false, richerror.New(op).
+			WithKind(richerror.KindUnexpected).
+			WithMessage(errmsg.CantScanQueryResult).
+			WithError(err)
+	}
+	return user, true, nil
+}
+
+func (d *DB) FindUserByEmail(ctx context.Context, email string) (userentity.UserWithPasswordHash, bool, error) {
+	const op = "postgresqluser.FindUserByEmail"
+	stmt, err := d.db.Conn().PrepareContext(ctx,
+		`SELECT username, email, password_hash FROM users WHERE email=$1`)
+
+	if err != nil {
+		return userentity.UserWithPasswordHash{}, false, richerror.New(op).
+			WithKind(richerror.KindUnexpected).
+			WithMessage(errmsg.CantScanQueryResult).
+			WithError(err)
+	}
+
+	row := stmt.QueryRowContext(ctx, email)
+
+	var user userentity.UserWithPasswordHash
+	err = row.Scan(&user.UserName, &user.Email, &user.PasswordHash)
```

Comment:
Turning this into a function + we should scan for user's mobile as well (we are returning the user's mobile in the method above)